File: branches/with-libpq-backend/src/dataModel/CustomSymbol.cpp
/*
 * File:   CustomSymbol.cpp
 * Author: jolo
 *
 * Created on 11. Dezember 2009, 16:37
 */

#include "CustomSymbol.h"

CustomSymbol::CustomSymbol(Project* p,
			   int id, 
			   const QString& name, 
			   const QString& description, 
			   const QString& fileName)
  : DatasetWithFileName(p,
			id, 
			name, 
			description, 
			fileName) {
}

CustomSymbol::~CustomSymbol() {
}

QString CustomSymbol::makeToolTipText(const bool withDatasetName) const {
  QStringList ret;
  if (withDatasetName) {
    ret << QObject::tr("Custom Symbol:");
  }

  ret << QObject::tr("Id: %1").arg(getId())
      << QObject::tr("Name: %1").arg(getName())
      << QObject::tr("Description: %1").arg(getDescription())
      << QObject::tr("File Name: %1").arg(getFileName());
  return ret.join("\n");
}

File: branches/experimental/src/Model/CustomSymbol.py
from NamedDescribedDatasetWithSvgItemInProject import NamedDescribedDatasetWithSvgItemInProject

class CustomSymbol(NamedDescribedDatasetWithSvgItemInProject):
    def __init__(self, project, id=None, name=None, svgItem=None, description=None, defaultGrainSize=None):
        super(CustomSymbol, self).__init__(project, id, name, svgItem, description)
